src/Microsoft.HttpRepl.IntegrationTests/Commands/PrefCommandTests.cs

```diff
@@ -0,0 +1,217 @@
+            HttpState httpState = new HttpState(fileSystem, preferences);
+            MockedShellState shellState = new MockedShellState();
+            PrefCommand command = new PrefCommand();
+
+            // First, set it to something other than the default and make sure that works.
+            string firstCommandExpectedValue = "BoldMagenta";
+            string firstCommandText = $"pref set {WellKnownPreference.ProtocolColor} {firstCommandExpectedValue}";
+            ICoreParseResult firstParseResult = CoreParseResultHelper.Create(firstCommandText);
+
+            await command.ExecuteAsync(shellState, httpState, firstParseResult, CancellationToken.None);
+
+            Assert.Empty(shellState.Output);
+            Assert.Equal(firstCommandExpectedValue, httpState.Preferences[WellKnownPreference.ProtocolColor]);
+
+            // Then, set it to nothing and make sure it goes back to the default
+            string secondCommandText = $"pref set {WellKnownPreference.ProtocolColor}";
+            ICoreParseResult secondParseResult = CoreParseResultHelper.Create(secondCommandText);
+
+            await command.ExecuteAsync(shellState, httpState, secondParseResult, CancellationToken.None);
+
+            Assert.Empty(shellState.Output);
+            Assert.Equal(httpState.DefaultPreferences[WellKnownPreference.ProtocolColor], httpState.Preferences[WellKnownPreference.ProtocolColor]);
+
+        }
+
+        [Fact]
+        public void CanHandle_NoArguments_ReturnsNull()
+        {
+            string commandText = "";
+
+            Arrange(commandText, out HttpState httpState, out MockedShellState shellState, out ICoreParseResult parseResult, out PrefCommand command);
+
+            bool? canHandle = command.CanHandle(shellState, httpState, parseResult);
+
+            Assert.Null(canHandle);
+        }
+
+        [Fact]
+        public void CanHandle_InvalidFirstArgument_ReturnsNull()
+        {
+            string commandText = "preferences set colors.protocol";
+
+            Arrange(commandText, out HttpState httpState, out MockedShellState shellState, out ICoreParseResult parseResult, out PrefCommand command);
+
+            bool? canHandle = command.CanHandle(shellState, httpState, parseResult);
+
+            Assert.Null(canHandle);
+        }
+
+        [Fact]
+        public void CanHandle_SetWithNoName_DisplaysError()
+        {
+            string commandText = "pref set";
+
+            Arrange(commandText, out HttpState httpState, out MockedShellState shellState, out ICoreParseResult parseResult, out PrefCommand command);
+
+            bool? canHandle = command.CanHandle(shellState, httpState, parseResult);
+
+            Assert.False(canHandle);
+            Assert.Equal(Resources.Strings.PrefCommand_Error_NoPreferenceName, shellState.ErrorMessage);
+        }
+
+        [Fact]
+        public void CanHandle_SetWithBlankName_DisplaysError()
```

Comment:
is the `parseResult` different between this test and the one above?  i.e. does the trailing space make a difference in the `parseResult`, or are these essentially the same tests by the time `CanHandle` is invoked?

Reply:
The parseResult is different in this case. In the SetWithNoName case, parseResult only has two sections (pref and set), so the check for # of arguments causes it to return false. In this case, parseResult has four sections (pref, set and two blank strings), so the check for whether or not the third argument is blank causes it to return false.